tests/e2e/vcp_to_csi_syncer.go

```diff
@@ -833,6 +833,154 @@ var _ = ginkgo.Describe("[csi-vcp-mig] VCP to CSI migration syncer tests", func(
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// Create file1.txt at mountpath inside pod.
+		ginkgo.By(fmt.Sprintf("Creating file file1.txt at mountpath inside pod: %v", pod.Name))
+		data1 := "This file file1.txt is written before migration"
+		filePath1 := "/mnt/volume1/file1.txt"
+		writeDataOnFileFromPod(namespace, pod.Name, filePath1, data1)
+
+		ginkgo.By("Enabling CSIMigration and CSIMigrationvSphere feature gates on kube-controller-manager")
+		err = toggleCSIMigrationFeatureGatesOnKubeControllerManager(ctx, client, true)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		kcmMigEnabled = true
+
+		ginkgo.By("Waiting for migration related annotations on PV/PVCs created before migration")
+		waitForMigAnnotationsPvcPvLists(ctx, client, vcpPvcsPreMig, vcpPvsPreMig, true, migrationEnabledByDefault)
+
+		ginkgo.By("Verify CnsVSphereVolumeMigration crds and CNS volume metadata on pvc created before migration")
+		verifyCnsVolumeMetadataAndCnsVSphereVolumeMigrationCrdForPvcs(ctx, client, vcpPvcsPreMig)
+
+		ginkgo.By("Enable CSI migration feature gates on kublets on k8s nodes")
+		toggleCSIMigrationFeatureGatesOnK8snodes(ctx, client, true, namespace)
+		kubectlMigEnabled = true
+
+		// Verify that fstype used is still xfs after migration
+		// verify that data can be read successfully that was written before migration
+		// Verify that new write is successful post migration
+		dep1, err = client.AppsV1().Deployments(namespace).Get(ctx, dep1.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		podsAfterMig, err := fdep.GetPodsForDeployment(client, dep1)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		podAfterMig := podsAfterMig.Items[0]
+		ginkgo.By("Verify if filesystem used to mount volume is xfs post migration")
+		_, err = framework.LookForStringInPodExec(namespace, podAfterMig.Name, []string{"/bin/cat", "/mnt/volume1/fstype"},
+			xfsFSType, time.Minute)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Verify that data can be successfully read from file1.txt which was written before migration")
+		output := readFileFromPod(namespace, podAfterMig.Name, filePath1)
+		gomega.Expect(output == data1+"\n").To(gomega.BeTrue(), "Pod is not able to read file1.txt post migration")
+
+		// Create new file file2.txt at mountpath inside pod.
+		ginkgo.By(fmt.Sprintf("Creating file file2.txt at mountpath inside pod: %v", podAfterMig.Name))
+		data2 := "This file file2.txt is written post migration"
+		filePath2 := "/mnt/volume1/file2.txt"
+		writeDataOnFileFromPod(namespace, podAfterMig.Name, filePath2, data2)
+
+		ginkgo.By("Verify that data written post migration can be successfully read from file2.txt")
+		output = readFileFromPod(namespace, podAfterMig.Name, filePath2)
+		gomega.Expect(output == data2+"\n").To(gomega.BeTrue(), "Pod is not able to read file2.txt")
+
+		ginkgo.By("Creating VCP PVC pvc2 post migration")
+		pvc2, err := createPVC(client, namespace, nil, "", vcpSc, "")
```

Comment:
we are not using this pvc anywhere right? shouldn't we use it with the deployment and verify the FS type?

Reply:
Added code to create new deployment to use this PVC, also verifying that fstype used is xfs.